addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -1223,58 +1223,58 @@ var VideoWidget = MediaWidget.extend({
 
-        var query = this._createVideoNode(url, {
+        this._createVideoNode(url, {
             'autoplay': this.isForBgVideo || this.$('input#o_video_autoplay').is(':checked'),
             'hide_controls': this.isForBgVideo || this.$('input#o_video_hide_controls').is(':checked'),
             'loop': this.isForBgVideo || this.$('input#o_video_loop').is(':checked'),
             'hide_fullscreen': this.isForBgVideo || this.$('input#o_video_hide_fullscreen').is(':checked'),
             'hide_yt_logo': this.isForBgVideo || this.$('input#o_video_hide_yt_logo').is(':checked'),
             'hide_dm_logo': this.isForBgVideo || this.$('input#o_video_hide_dm_logo').is(':checked'),
             'hide_dm_share': this.isForBgVideo || this.$('input#o_video_hide_dm_share').is(':checked'),
-        });
-
-        var $optBox = this.$('.o_video_dialog_options');
-
-        // Show / Hide preview elements
-        this.$el.find('.o_video_dialog_preview_text, .media_iframe_video_size').add($optBox).toggleClass('d-none', !query.$video);
-        // Toggle validation classes
-        this.$el.find('#o_video_form_group')
-            .toggleClass('o_has_error', !query.$video).find('.form-control, .custom-select').toggleClass('is-invalid', !query.$video)
-            .end()
-            .toggleClass('o_has_success', !!query.$video).find('.form-control, .custom-select').toggleClass('is-valid', !!query.$video);
-
-        // Individually show / hide options base on the video provider
-        $optBox.find('div.o_' + query.type + '_option').removeClass('d-none');
-
-        // Hide the entire options box if no options are available or if the
-        // dialog is opened for a background-video
-        $optBox.toggleClass('d-none', this.isForBgVideo || $optBox.find('div:not(.d-none)').length === 0);
-
-        if (query.type === 'yt') {
-            // Youtube only: If 'hide controls' is checked, hide 'fullscreen'
-            // and 'youtube logo' options too
-            this.$('input#o_video_hide_fullscreen, input#o_video_hide_yt_logo').closest('div').toggleClass('d-none', this.$('input#o_video_hide_controls').is(':checked'));
-        }
+        }).then((query) => {
+            var $optBox = this.$('.o_video_dialog_options');
+
+            // Show / Hide preview elements
+            this.$el.find('.o_video_dialog_preview_text, .media_iframe_video_size').add($optBox).toggleClass('d-none', !query.$video);
+            // Toggle validation classes
+            this.$el.find('#o_video_form_group')
+                .toggleClass('o_has_error', !query.$video).find('.form-control, .custom-select').toggleClass('is-invalid', !query.$video)
+                .end()
+                .toggleClass('o_has_success', !!query.$video).find('.form-control, .custom-select').toggleClass('is-valid', !!query.$video);
+
+            // Individually show / hide options base on the video provider
+            $optBox.find('div.o_' + query.type + '_option').removeClass('d-none');
```

Comment:
here we do not change anything https://github.com/odoo/odoo/pull/44537/files#diff-194923481b66ad4811cdca3dc7b96693R1199  that's why we use 'query.type' instead of 'query.plateform'